x/swaprouter/router_test.go

```diff
@@ -159,3 +164,68 @@ func (suite *KeeperTestSuite) TestMultihopSwapExactAmountOut() {
 		})
 	}
 }
+
+// TestGetSwapModule tests that the correct swap module is returned for a given pool id.
+// Additionally, validates that the expected errors are produced when expected.
+func (suite *KeeperTestSuite) TestGetSwapModule() {
+
+	tests := map[string]struct {
+		poolId            uint64
+		preCreatePoolType types.PoolType
+		routesOverwrite   map[types.PoolType]types.SwapI
+
+		expectedModule reflect.Type
+		expectError    error
+	}{
+		"valid balancer pool": {
+			preCreatePoolType: types.Balancer,
+			poolId:            1,
+			expectedModule:    gammKeeperType,
+		},
+		"non-existent pool": {
+			preCreatePoolType: types.Balancer,
+			poolId:            2,
+			expectedModule:    gammKeeperType,
+
+			expectError: types.FailedToFindRouteError{PoolId: 2},
+		},
+		"undefined route": {
+			preCreatePoolType: types.Balancer,
+			poolId:            1,
+			routesOverwrite: map[types.PoolType]types.SwapI{
+				types.StableSwap: &gamm.Keeper{}, // undefined for balancer.
+			},
+
+			expectError: types.UndefinedRouteError{PoolId: 1, PoolType: types.Balancer},
+		},
+		// TODO: valid stableswap test case.
+		// TODO: valid concentrated liquidity test case.
+	}
+
+	for name, tc := range tests {
+		tc := tc
+		suite.Run(name, func() {
+			suite.SetupTest()
+			swaprouterKeeper := suite.App.SwapRouterKeeper
+
+			suite.createPoolFromType(tc.preCreatePoolType)
```

Comment:
this seems odd, can we just create a single pool prior to running the test cases? from the looks of it, it feels like we should be creating a second balancer pool here with poolId 2 and should not fail (I know thats not how this works, but just is how it looks like it should work on first glance)

Reply:
Why?

We need to test all pool types eventually. We also need to be able to control which kind of pool we are creating from the test case parameter. This helps to achieve that.